### DISCUSSION TITLE
Saving display preferences per monitor

### DISCUSSION THREAD

**Discussion by MattHorn25** [Albeoris/Memoria#946] [Ideas]
Hello!

This is something I've run across while playing the game in handheld mode and then docked mode on my Steam Deck: the launcher will always remember and present the display settings from last time, even if the current Active (and only) Monitor doesn't support these particular settings.

It seems like a feature where the launcher saves the following settings:

- Window Mode
- Resolution
- Widescreen toggle
- FPS

and also potentially these (debatable):

- Battle HUD
- UI lines/columns

for the currently active monitor, and swaps them out if a previously configured monitor is detected as Active, would be beneficial.

Thanks for everything you guys do!

**Comment by SamsamTS** [comment]
Does resolution change between docked/handheld mode. Is this the main issue? That's really the only setting that would potentially not be supported and possibly lead to a cropped window in borderless fullscreen or windowed. Frame rate isn't really an issue since it's the internal frame rate not the monitor refresh rate.

I added a check that will limit the resolution to the maximum the current display is capable of, so hopefully that will help. You will have to report back with the next release.

**Reply by MattHorn25** [reply]
Thanks! For the resolution, yes it changes between modes - since the Deck's handheld display is only 1280x800, any external display will almost always have a higher resolution. The external display's reported resolution can be set to more or less anything (from what I understand the Deck virtualizes peripherals to an extent), but 1080p is a common setting.

And noted on the framerate clarification, although I'm not sure whether a mismatch would result in any stutter? This is part of why I thought per-screen settings might be useful.

**Reply by SamsamTS** [reply]
There will be presets with the new launcher. You will be able to save your settings as a preset if you really want different options for handheld/docked. Unfortunately presets cannot include screen mode, resolution or monitor but everything else can be saved. Might be worth considering.
